lightningd/peer_control.c

```diff
@@ -176,6 +176,55 @@ u8 *p2wpkh_for_keyidx(const tal_t *ctx, struct lightningd *ld, u64 keyidx)
 	return scriptpubkey_p2wpkh(ctx, &shutdownkey);
 }
 
+static struct simple_htlc **collect_htlcs(struct channel *channel, u32 local_feerate) {
+	// Collect the htlcs for call to hsmd.
+	struct htlc_in_map *htlcs_in = &channel->peer->ld->htlcs_in;
+	struct htlc_out_map *htlcs_out = &channel->peer->ld->htlcs_out;
+	struct simple_htlc **htlcs = tal_arr(tmpctx, struct simple_htlc *, 0);
+
+	const struct htlc_in *hin;
+	struct htlc_in_map_iter ini;
+	for (hin = htlc_in_map_first(htlcs_in, &ini);
+	     hin;
+	     hin = htlc_in_map_next(htlcs_in, &ini)) {
+		if (hin->key.channel != channel)
+			continue;
+		if (htlc_is_trimmed(REMOTE, hin->msat, local_feerate,
+				    channel->our_config.dust_limit, LOCAL,
+				    channel_has(channel, OPT_ANCHOR_OUTPUTS)))
+			continue;
+		struct simple_htlc *simple =
+		    new_simple_htlc(NULL,
+				    htlc_state_owner(hin->hstate),
+				    hin->msat,
+				    &hin->payment_hash,
+				    hin->cltv_expiry);
+		tal_arr_expand(&htlcs, tal_steal(htlcs, simple));
+	}
+	const struct htlc_out *hout;
+	struct htlc_out_map_iter outi;
+	for (hout = htlc_out_map_first(htlcs_out, &outi);
+	     hout;
+	     hout = htlc_out_map_next(htlcs_out, &outi)) {
+		if (hout->key.channel != channel)
+			continue;
+		if (hout->hstate < RCVD_ADD_REVOCATION || hout->hstate > RCVD_REMOVE_COMMIT)
+			continue;
```

Comment:
I'm not sure why we don't need the equivalent filter on `htlcs_in` above ...

Reply:
did you try to add the filter (with `s/RCVD/SENT/`)?  does it fail?

Reply:
We removed this code w/ the `sign_holder_commitment_tx` API cleanup